Makefile:  failed to compile without git repository

<!-- Please use this template while reporting an issue, providing as much information as possible. Failure to do so may result in a delayed response. Thank you! -->


**Description of problem:**
Download and use GlusterFS zip pakage. 
When compile it without git repository, a git error will fail the make.

**The exact command to reproduce the issue**:
./autogen.sh
./configure
make dist

**The full output of the command that failed**:
(cd . && git diff && echo ===== git log ==== && git log) > glusterfs-8.2/ChangeLog
usage: git diff [--no-index] <path> <path>
make[2]: *** [gen-ChangeLog] Error 129
<details>

**Expected results:**
<details>


**Mandatory info:**
**- The output of the `gluster volume info` command**:

**- The output of the `gluster volume status` command**:

**- The output of the `gluster volume heal` command**:

**- Provide logs present on following locations of client and server nodes -
/var/log/glusterfs/

**- Is there any crash ? Provide the backtrace and coredump


**Additional info:**

</details>

**- The operating system / glusterfs version**:

**Note: Please hide any confidential data which you don't want to share in public like IP address, file name, hostname or any other configuration**

--- a/Makefile.am
+++ b/Makefile.am
@@ -44,7 +44,9 @@ clang-check:
 	@$(top_srcdir)/extras/clang-checker.sh
 
 gen-ChangeLog:
-	(cd $(srcdir) && git diff && echo ===== git log ==== && git log) > $(distdir)/ChangeLog
+	if test -d $(srcdir)/.git; then            	\
+	  (cd $(srcdir) && git diff && echo ===== git log ==== && git log) > $(distdir)/ChangeLog; \
+	fi
 
 .PHONY : gen-VERSION
 gen-VERSION:

